src/libraries/Common/src/Roslyn/SyntaxValueProvider_ForAttributeWithMetadataName.cs

```diff
@@ -113,49 +113,39 @@ internal static partial class SyntaxValueProviderExtensions
 
 #endif
 
-        var collectedNodes = nodesWithAttributesMatchingSimpleName
-            .Collect()
-            .WithComparer(ImmutableArrayValueComparer<SyntaxNode>.Instance)
-            /*.WithTrackingName("collectedNodes_ForAttributeWithMetadataName")*/;
-
-        // Group all the nodes by syntax tree, so we can process a whole syntax tree at a time.  This will let us make
-        // the required semantic model for it once, instead of potentially many times (in the rare, but possible case of
-        // a single file with a ton of matching nodes in it).
-        var groupedNodes = collectedNodes.SelectMany(
-            static (array, cancellationToken) =>
-                array.GroupBy(static n => n.SyntaxTree)
-                     .Select(static g => new SyntaxNodeGrouping<SyntaxNode>(g)))/*.WithTrackingName("groupedNodes_ForAttributeWithMetadataName")*/;
```

Comment:
previously, the approach was that we would flatten all found nodes across all files into one big list.  we'd then regroup those per file here so we could process all the nodes in the file at the same time with a single semantic model.

The new approach avoids this pointless flattening/regrouping.  The initial pass already is producing all matching nodes per tree.  so it can just return things in that grouping anyways.   this save two intermediary tables and lots of work to transform between things.